Fixes to multiple failing E2E tests - trunk mirror from #10057

alopezari opened the pull request:
This PR contains the same changes as #10057 but pointing to `trunk`. I decided to create 2 different PRs for `trunk` and `develop` because the original PR was up to date with develop and contained recent changes that conflicted when updating it for `trunk`. In order to avoid conflicts and keeping things simple, this new PR was created to enable these changes in `trunk`.

---
Fixes #9746 
Fixes #10115

#### Changes proposed in this Pull Request

This PR fixes the following failing E2E tests:

- **tests/e2e/specs/wcpay/merchant/merchant-payment-gateways-confirmation.spec.js:**
These were failing because of wrong locator that seemed to have been outdated at some point.

- **tests/e2e/specs/wcpay/shopper/shopper-checkout-failures.spec.js:**
Same as before, the tests were failing due to inaccurate locators.

- **tests/e2e/specs/wcpay/shopper/shopper-pay-for-order.spec.js:**
Inaccurate locators again.

- **Shopper tests in WooCommerce Beta version**:
[WooCommerce 9.6 enables by default the Coming Soon mode for new sites](https://github.com/woocommerce/woocommerce/pull/53685), which breaks our tests. In order to bypass this, the script that sets up our test environment will deactivate this mode by disabling a flag in the Options table.

- **tests/e2e/specs/subscriptions/merchant/merchant-subscriptions-renew-action-scheduler.spec.js**:
[WooCommerce 9.6 adds a new logic to define which fields are shown by default in the checkout form](https://github.com/woocommerce/woocommerce/pull/52784), and [the `company` field](https://github.com/woocommerce/woocommerce/pull/52784/files#diff-73d3378d53fd116ec265503878cd4bdddb12396568757b283cb86303be8033a1R38) from the billing section will now be hidden by default, unless you enable it. We've updated our test environment setup flow to enable the `company` field - used by our tests - in the checkout form by enabling a flag in the Options table.

- **tests/e2e/specs/wcpay/shopper/shopper-multi-currency-widget.spec.js**:
There seems to be a race condition here that make a test from this spec fail on CI but pass on my local environment, so I can't debug it properly. After spending some time with this one, I've decided to skip this test because it will be migrated soon and we don't need to spend some more extra time on it.

<!--
Title: A descriptive, yet concise, title.
-->

<!--
Description: Write a brief summary about this PR. As you compose your summary, consider each of these questions and address them if appropriate. Why is this change needed? What does this change do? Were there other solutions you considered? Why did you choose to pursue this solution? Describe any trade-offs you might have had to make.
-->

<!--
Questions for PR author:
- How can this code break?
- What are we doing to make sure this code doesn't break?
-->

<!--
Images or gifs: Include before and after screenshots or gifs/videos when it makes sense.
-->

#### Testing instructions

<!--
Testing instructions: How should this be tested and how can a reviewer test the end-user functionality? Are there known issues that you plan to address in a future PR? Are there any side effects that readers should be aware of?
-->

<!--
Add as many details as possible to help others reproduce the issue and test the fix.
"Before / After" screenshots can also be very helpful when the change is visual.
-->

* Check that all checks passed for this PR.
* Note that the [E2E Tests - All](https://github.com/Automattic/woocommerce-payments/actions/runs/12711344076) workflow passed for this branch.
* Set up the E2E env with by following the steps described [here](https://github.com/Automattic/woocommerce-payments/tree/develop/tests/e2e#setting-up--running-e2e-tests).
* Run the `merchant-payment-gateways-confirmation` tests in headless mode with `npm run test:e2e tests/e2e/specs/wcpay/merchant/merchant-payment-gateways-confirmation.spec.js`.
* Run the `merchant-payment-gateways-confirmation` tests in UI mode with `npm run test:e2e-dev tests/e2e/specs/wcpay/merchant/merchant-payment-gateways-confirmation.spec.js`.
* Run the `shopper-checkout-failures` tests in headless mode with `npm run test:e2e tests/e2e/specs/wcpay/shopper/shopper-checkout-failures.spec.js`.
* Run the `shopper-checkout-failures` tests in UI mode with `npm run test:e2e-dev tests/e2e/specs/wcpay/shopper/shopper-checkout-failures.spec.js`.
* Run the `shopper-pay-for-order` tests in headless mode with `npm run test:e2e tests/e2e/specs/wcpay/shopper/shopper-pay-for-order.spec.js`.
* Run the `shopper-pay-for-order` tests in UI mode with `npm run test:e2e-dev tests/e2e/specs/wcpay/shopper/shopper-pay-for-order.spec.js`.
* Run the `shopper-multi-currency-widget` tests in headless mode with `npm run test:e2e tests/e2e/specs/wcpay/shopper/shopper-multi-currency-widget.spec.js`.
* Run the `shopper-multi-currency-widget` tests in UI mode with `npm run test:e2e-dev tests/e2e/specs/wcpay/shopper/shopper-multi-currency-widget.spec.js`.
* Reset the environment with `npm run test:e2e-reset`.
* Add `E2E_WC_VERSION=beta` to your `local.env` file, this will configure your environment to use the latest WooCommerce Beta version.
* Set up the environment again with `npm run test:e2e-setup`.
* Run the `merchant-orders-status-change` tests in headless mode with `npm run test:e2e tests/e2e/specs/wcpay/merchant/merchant-orders-status-change.spec.js`.
* Run the `merchant-orders-status-change` tests in UI mode with `npm run test:e2e-dev tests/e2e/specs/wcpay/merchant/merchant-orders-status-change.spec.js`.
* Run the `merchant-subscriptions-renew-action-scheduler` tests in headless mode with `npm run test:e2e tests/e2e/specs/subscriptions/merchant/merchant-subscriptions-renew-action-scheduler.spec.js`.
* Run the `merchant-subscriptions-renew-action-scheduler` tests in UI mode with `npm run test:e2e-dev tests/e2e/specs/subscriptions/merchant/merchant-subscriptions-renew-action-scheduler.spec.js`.
* Finally, restore your test environment to use the latest standard WooCommerce version with `npm run test:e2e-reset` and removing `E2E_WC_VERSION=beta` from your `local.env` file.
-------------------

- [ ] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

<!--
Make sure you edit the page for the current release when adding testing instructions.
We often create a blank page ahead of time for the next release.
If this PR need not be QA tested, edit to 'QA Testing Not Applicable'
-->

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

alopezari committed:
Fixes to multiple failing E2E tests - trunk mirror from #10057.

botwoo commented:
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `10140` or branch name `fix/failing-e2e-tests-sync-trunk` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=fix/failing-e2e-tests-sync-trunk) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 5be2bc651558a2889cf213e59d01f4b24bbd1469
- Build time: 2025-01-14 08:54:22 UTC

_Note: the build is updated when a new commit is pushed to this PR._

achyuthajoy reviewed (commented):
@alopezari Changes look good to me, but I'm seeing failures on WCPay - Merchant tests when running tests against this branch - https://github.com/Automattic/woocommerce-payments/actions/runs/12762605261

Could you check?

alopezari commented:
Thanks for the review @achyuthajoy! I remember I fixed those tests on `develop` but then some specs were deleted as part of the migration to Playwright, so I deleted these specs on the branch as my fixes conflicted with the deleted Puppeteer specs.

Then, I replicated the changes from the branch that contains the fixes for `develop` into this one for `trunk`, so those fixes are missing here.

~I can add the fixes here to keep the workflow green or we can take the results of `E2E tests - All` against `develop` as a reference, since eventually those changes should arrive to `trunk`. What do you prefer?~

**Update: The fix is pretty easy, I can add it anyway and keep things clean** 👍

alopezari committed:
Fix tests from merchant-payment-gateways-confirmation and shopper-pay-for-order.

alopezari commented:
Hi @achyuthajoy!

Added the changes and the tests passed:
https://github.com/Automattic/woocommerce-payments/actions/runs/12764357695

alopezari commented:
Closing this as it will be handled in a different way.
